c7n/resources/awslambda.py

```diff
@@ -644,10 +644,13 @@ class LayerCrossAccount(CrossAccountAccessFilter):
     def process(self, resources, event=None):
         client = local_session(self.manager.session_factory).client('lambda')
         for r in resources:
-            r['c7n:Policy'] = self.manager.retry(
-                client.get_layer_version_policy,
-                LayerName=r['LayerName'],
-                VersionNumber=r['Version']).get('Policy')
+            if 'c7n:Policy' not in r:
```

Comment:
general preference would be to try and minimize nesting (cyclomatic complexity).

ie restructuring

```python
if 'c7n:Policy' in r:
   continue
```